### PR TITLE
add error message if initial column isn't selected

### LINKED ISSUE #106: Format-data: add alert for join/split button if no initial field selection has been made

### PR DESCRIPTION
**Pull request by meghanhade**
This adds an error message when the user tries to click `Join columns` or `Split column` without selecting a column from the initial dropdown.

In the `select-column` component, I switched the `chooseField` action to sending that action to the data-format controller, so that the error state could be cleared when a column is selected.

Closes #106

### CODE REVIEW

**Comment by hanbyul-here** [comment]
Is the issue related to the last comment on https://github.com/mapzen/mobility-on-demand-sandbox-contributions/issues/42 ?

**Comment by meghanhade** [comment]
@hanbyul-here yep, sorry about that! I checked of the item this handles in #42. This doesn't require validation, so I thought a separate one-off PR would be ok.

**Comment by hanbyul-here** [comment]
yeah I like separate issue better. Thanks @meghanhade

File: app/controllers/data-format.js [hunk 1]
```diff
@@ -43,20 +43,42 @@ export default Ember.Controller.extend(sharedActions, {
...
-      Ember.set(this.model.submission.get('oaFields')[field], "function", action);
-      if (action === "join"){
-        this.set('showAdditionalJoinDropdown', true);
+      if (this.model.submission.get('oaFields')[field].fields.length > 0){
+        this.resetErrorState();
+        Ember.set(this.model.submission.get('oaFields')[field], "function", action);
+        if (action === "join"){
+          this.set('showAdditionalJoinDropdown', true);
+        }
+      } else {
```
Comment (hanbyul-here) [inline app/controllers/data-format.js]: How many error catches are we going to need for this component? If there is more than one, we maybe want to separate the function from whole process to make it clear where error-catching is happening.

**Comment by hanbyul-here** [comment]
This one look good, well working. I will let you decide on  how we want to deal with error catching (I left a comment).

**Comment by meghanhade** [comment]
Thanks @mkong0216!

**Pull request merged**